shared/utils/typedefs-wrappers.hpp

```diff
@@ -563,6 +565,92 @@ struct SafePtrUnity : public SafePtr<T, true> {
     }
 };
 
+template <typename T>
+#ifdef HAS_CODEGEN
+    requires(std::is_assignable_v<UnityEngine::Object, T>)
+#endif
+struct UnityW {
+    UnityW(T* t) : innerPtr(t) {}
+    UnityW() = default;
+    UnityW(void* p) : innerPtr(p) {}
+
+    T const* unsafePtr() const {
+        return innerPtr;
+    }
+
+    T* unsafePtr() {
+        return innerPtr;
+    }
+
+    T* ptr() {
+        __SAFE_PTR_UNITY_NULL_HANDLE_CHECK(innerPtr);
+    }
+
+    T const* ptr() const {
+        __SAFE_PTR_UNITY_NULL_HANDLE_CHECK(innerPtr);
+    }
+
+    constexpr void* convert() {
```

Comment:
convert has to be implemented as 
```c++
constexpr void* convert() const noexcept {
    return const_cast<void*>(static_cast<const void*>(inner_ptr));
}
```
to work properly with all sorts of our utilities (since we can't really guarantee const-ness anyway on the boundary between mods and il2cpp)

Reply:
also I suggest using unsafePtr() here instead since convert() by itself shouldn't be throwing, converting from a wrapped nullptr instance should be allowed